src/System.Threading.Tasks/tests/CancellationTokenTests.cs

```diff
@@ -817,6 +817,12 @@ public static void ODEWhenDisposingLinkedCTS()
             }
         }
 
+        [System.Runtime.CompilerServices.MethodImplAttribute(System.Runtime.CompilerServices.MethodImplOptions.NoInlining)]
```

Comment:
I'm confused about the impact of your GC change.  In CoreFx debug builds, this method shouldn't get inlined, as optimizations including inlining would be disabled, right?  Which if true suggests that this NoInlining is here to prevent this from being inlined in release.  That means your change could end up causing objects to be preserved to the end of a method in release?  We're not worried about such a change in behavior?